[FLINK-11635][docs-zh] translate dev/stream/state/checkpointing into Chinese

Henvealf opened the pull request:
Translate the checkpointing doc into Chinese。

wuchong commented:
Hi @Henvealf , this is not a pull request for FLINK-14227. Please correct it in the title. Thanks.

yangjf2019 commented:
Hi @Henvealf ,Thanks for your effort. [here](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) are some Flink Translation Specifications .Maybe it is useful to you.

KarmaGYZ commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 
-In order to make state fault tolerant, Flink needs to **checkpoint** the state. Checkpoints allow Flink to recover state and positions
-in the streams to give the application the same semantics as a failure-free execution.
+[Documentation on streaming fault tolerance]({{ site.baseurl }}/internals/stream_checkpointing.html) 介绍了 Flink 流计算容错机制的内部技术原理。
```
Shoule be `{{ site.baseurl }}/zh/internals`

KarmaGYZ commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
+    该选项在检查点间的最小时间被定义时不能使用。
+    
+  - *外部检查点（externalized checkpoints）*: 你可以配置周期检查点存储到外部系统中。外部检查点将他们的元数据写到持久化存储上并且在 job 失败的时候*不会*被自动删除。
+    这种方式下，如果你的 job 失败，你将会有一个现有的检查点去恢复。更多的细节请看 [deployment notes on externalized checkpoints]({{ site.baseurl }}/ops/state/checkpoints.html#externalized-checkpoints)。
```
Same as above `{{ site.baseurl }}/zh/ops`

KarmaGYZ commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
+### 相关的配置选项
 
-Some more parameters and/or defaults may be set via `conf/flink-conf.yaml` (see [configuration]({{ site.baseurl }}/ops/config.html) for a full guide):
+更多的属性 与/或 默认值能在 `conf/flink-conf.yaml` 中设置（完整教程请阅读 [configuration]({{ site.baseurl }}/ops/config.html)）。
```
Same as above.

Henvealf commented:
> Hi @Henvealf , this is not a pull request for FLINK-14227. Please correct it in the title. Thanks.

Which title should I use, FLINK-14313 ?

wuchong commented:
I think it should be FLINK-11635. I updated the title.

Hi @xintongsong , do you have other concerns on this?

xintongsong commented:
Thanks @wuchong, I do have other concerns.

I've left several comments, regarding translations that IMO are not easy to understand. These comments are marked 'resolved' by @Henvealf without addressed or any further explanation.

Henvealf commented:
> Thanks @wuchong, I do have other concerns.
> 
> I've left several comments, regarding translations that IMO are not easy to understand. These comments are marked 'resolved' by @Henvealf without addressed or any further explanation.

I have commit the update.

xintongsong commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 env.getCheckpointConfig().enableExternalizedCheckpoints(ExternalizedCheckpointCleanup.RETAIN_ON_CANCELLATION);
 
-// allow job recovery fallback to checkpoint when there is a more recent savepoint
+// 允许在有更近保存点时回退到 checkpoint
```
savepoint不翻译

xintongsong commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 env.get_checkpoint_config().enable_externalized_checkpoints(ExternalizedCheckpointCleanup.RETAIN_ON_CANCELLATION)
 
-# allow job recovery fallback to checkpoint when there is a more recent savepoint
+# 允许在有更近保存点时回退到 checkpoint
```
savepoint不翻译

Henvealf commented:
> Thank for addressing the comments, @Henvealf.
> I find two occurrences of 'savepoint' which is suggested not to be translated according to the translation specifications. Besides that, I have no other concerns.

Thanks! I have fixed this.

Henvealf commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 
-In order to make state fault tolerant, Flink needs to **checkpoint** the state. Checkpoints allow Flink to recover state and positions
-in the streams to give the application the same semantics as a failure-free execution.
+[Documentation on streaming fault tolerance]({{ site.baseurl }}/zh/internals/stream_checkpointing.html) 介绍了 Flink 流计算容错机制的内部技术原理。
```
好

Henvealf commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 val env = StreamExecutionEnvironment.getExecutionEnvironment()
 
-// start a checkpoint every 1000 ms
+// 每 1000 ms 开始一个 checkpoint
```
改成了 “每 1000ms 开始一次 checkpoint”

Henvealf commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 
-// checkpoints have to complete within one minute, or are discarded
+// Checkpoint 必须在一分钟内完成，否则就会被丢弃
 env.getCheckpointConfig.setCheckpointTimeout(60000)
```
丢弃是什么都不剩，放弃可能是不使用了，但是还会遗留写东西。我已经改成了“放弃”，不知哪个更符合实现。

wuchong commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
+  - *Checkpoint超时（checkpoint timeout）*：如果过了这个时间，还在进行中的 checkpoint 操作就会被抛弃。
+  
+  - *checkpoint 之间的最小时间（minimum time between checkpoints）*： 为了确保流应用在 checkpoint 之间有足够的进展，可以定义在 checkpoint 之间需要多久的时间。如果值设置为了 *5000*，
+    无论 checkpoint 持续时间与间隔是多久，在前一个 checkpoint 完成的五秒后才会开始下一个 checkpoint。
```
```suggestion
    无论 checkpoint 持续时间与间隔是多久，在前一个 checkpoint 完成的至少五秒后会才开始下一个 checkpoint。
```
这里 `no sooner than 5 seconds` 很重要，我觉得得翻译出来。

klion26 commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
+  - *精确一次（exactly-once） 对比 至少一次（at-least-once）*：你可以选择向 `enableCheckpointing(n)` 方法中传入一个模式来选择使用两种保证等级中的哪一种。
+    对于大多数应用来说，精确一次是较好的选择。至少一次可能与某些延迟超低（始终只有几毫秒）的应用的关联较大。
+  
+  - *Checkpoint超时（checkpoint timeout）*：如果过了这个时间，还在进行中的 checkpoint 操作就会被抛弃。
```
```suggestion
  - *Checkpoint 超时（checkpoint timeout）*：如果过了这个时间，还在进行中的 checkpoint 操作就会被抛弃。
```

wuchong commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
-  - A persistent storage for state, typically a distributed filesystem (e.g., HDFS, S3, GFS, NFS, Ceph, ...)
+## 激活与配置 Checkpoint
 
+默认情况下，Checkpoint 是禁用的。通过调用 `StreamExecutionEnvironment` 的 `enableCheckpointing(n)` 来激活 Checkpoint，里面的 *n* 是进行 Checkpoint 的间隔，单位毫秒。
```
+1 to "启用"

wuchong commented:
Hi @Henvealf  do you have time to update this?

Henvealf commented:
> Hi @Henvealf do you have time to update this?

Sorry for not replying for a long time, I will update it tomorrow.

Henvealf commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -25,146 +25,138 @@ under the License.
 env.getCheckpointConfig().setMinPauseBetweenCheckpoints(500);
 
-// checkpoints have to complete within one minute, or are discarded
+// Checkpoint 必须在一分钟内完成，否则就会被放弃
```
所有"弃" 都统一改成了 “抛弃”。

Henvealf commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -173,30 +165,26 @@ Some more parameters and/or defaults may be set via `conf/flink-conf.yaml` (see
-
-Flink supports different restart strategies which control how the jobs are restarted in case of a failure. For more 
-information, see [Restart Strategies]({{ site.baseurl }}/dev/restart_strategies.html).
+Flink 支持不同的重启策略，来控制 job 万一故障时该如何重启。更多信息请阅读 [Restart Strategies]({{ site.baseurl }}/zh/dev/restart_strategies.html)。
```
看了下会跳转。

Henvealf commented on docs/dev/stream/state/checkpointing.zh.md:
```diff
@@ -173,30 +165,26 @@ Some more parameters and/or defaults may be set via `conf/flink-conf.yaml` (see
 
 ## Selecting a State Backend
```
OK
